With $M, r, y_0 > 0$, the logistic difference equation is defined, For $n \in \mathbb R$. 

as $y_n := y_n-_1 + ry_n-_1(M - y_n-_1).$

$M_b:=48000$

$y_0=1$ for $n\in \mathbb R$

$y_n$ is the predicted number of infected individuals in the $buffalo \ \  population$ after $n$ days

$Question \  4.1$

In [10]:
def logistic_non_recursive(n, r, M):
    K=[1]
    for v in range(1, n+1):
        K.append(K[v-1]+r*K[v-1]*(M-K[v-1]))
    
    return K[n]

In [11]:
r_with_preventative_measures=6.7*10**(-7)

In [12]:
r_without_preventative_measures=2.1*10**(-6)

In [13]:
M=48000

In [14]:
logistic_non_recursive(0,r_with_preventative_measures , M)

1

In [15]:
logistic_non_recursive(1,r_with_preventative_measures , M)

1.03215933

In [16]:
logistic_non_recursive(3,r_with_preventative_measures , M)

1.0996138478181348

In [17]:
logistic_non_recursive(3,r_without_preventative_measures , M)

1.333897683083141

$Question \ 4.2$

Calculating the first $n$ For $y_n ≥ 48000/2$ with preventative mearsures

$Let \  n =x$

In [18]:
M=48000
r= 6.7*10**(-7)

y = lambda x:(x-1) + 𝑟*(x-1)*(𝑀-(x-1))
import sympy
import math 
from sympy.abc import x

In [19]:
y(x)

x + (48001 - x)*(6.7e-7*x - 6.7e-7) - 1

In spimplified form $y:=(-6.7/10^7)x^2 +1.03216737x -1.03216067$

For $y≥48000/2$, Similarly $y-48000/2≥0$..THerefore:

$(-6.7/10^7)x^2 + 1.03216737x - 24001.03216 ≥ 0$

In [20]:
a=-1.03216737
k=-6.7
p=10**7
b=4*(k/p)*(-24001.03216)
c=2*(k/p)
d=(a)**2
e=math.sqrt(d-b)

In [21]:
First_n=(a-e)/c

First_n

1516933.2760242098

In [22]:
Second_n=(a+e)/c

In [23]:
Second_n

23615.03740862616

Therefore the first value of $n ≥ 23615$ but $n≤1516934$

which means $23615≤n≤1516934$

In [24]:
logistic_non_recursive(23615, r, M)

47999.99999999989

$Question \ 4.3$


Calculating the first $n$ For $y_n ≥ 48000/2$ whithout preventative measures 

$Let \  n =x$

In [25]:
M=48000
r= 2.1*10**(-6)

y = lambda x:(x-1) + r*(x-1)*(𝑀-(x-1))
import sympy
import math 
from sympy.abc import x

In [26]:
y(x)

x + (48001 - x)*(2.1e-6*x - 2.1e-6) - 1

In spimplified form $y:=(-2.1/10^6)x^2 +1.1008042x -1.1008021$

For $y≥48000/2$, Similarly $y-48000/2≥0$..THerefore:

$(-2.1/10^6)x^2 + 1.1008042x - 24001.1008 ≥ 0$

In [27]:
o=-1.1008042
f=4*(-2.1/10**6)*(-24001.1008)
z=2*(-2.1/10**6)
s=(o)**2
q=math.sqrt(d-b)

In [28]:
First_N=(o+q)/z

In [29]:
Second_N=(o-q)/z

In [30]:
First_N

23876.423839895007

In [31]:
Second_N

500316.05235058116

Therefore the first value of $n ≥ 23876$ but $n≤500316$

which means $23876≤n≤500316$